src/annotator/anchoring/test/types-test.js

```diff
@@ -19,43 +21,290 @@ describe('types', function () {
+      });
+    });
+
+    describe('#toSelector', () => {
+      it('returns a TextPositionSelector', () => {
+        const anchor = createTextPositionAnchor();
+        assert.deepEqual(anchor.toSelector(), {
+          type: 'TextPositionSelector',
+          start: 0,
+          end: 0,
+        });
+      });
+    });
+
+    describe('#toRange', () => {
+      it('returns a range object', () => {
+        const anchor = createTextPositionAnchor();
+        fakePosToRange.returns('fake range');
+        assert.equal(anchor.toRange(), 'fake range');
+        assert.calledWith(fakePosToRange, container, { start: 0, end: 0 });
+      });
+    });
+  });
+
+  describe('TextQuoteAnchor', () => {
+    let fakeQuoteToRange;
+    let fakeQuoteFromRange;
+    let fakeToTextPosition;
+
+    beforeEach(() => {
+      fakeQuoteToRange = sinon.stub();
+      fakeQuoteFromRange = sinon.stub();
+      fakeToTextPosition = sinon.stub();
+      $imports.$mock({
+        'dom-anchor-text-quote': {
+          fromRange: fakeQuoteFromRange,
+          toRange: fakeQuoteToRange,
+          toTextPosition: fakeToTextPosition,
+        },
+      });
+    });
+
+    afterEach(() => {
+      $imports.$restore();
+    });
+
+    function createTextQuoteAnchor() {
+      return new TextQuoteAnchor(container, 'Liberty', {
+        prefix: 'a new nation, conceived in ',
+        suffix: ', and dedicated to the proposition that',
+      });
+    }
+
+    it('creates a TextQuoteAnchor if `context` is missing', () => {
+      assert.instanceOf(
+        new TextQuoteAnchor(container, 'Liberty'),
+        TextQuoteAnchor
+      );
+    });
+
+    describe('#fromRange', () => {
+      it('returns a range object', () => {
+        fakeQuoteFromRange.returns({
+          prefix: 'Four score and ',
+          suffix: 'brought forth on this continent',
+        });
+        const anchor = TextQuoteAnchor.fromRange(container, new Range());
+        assert.called(fakeQuoteFromRange);
+        assert.instanceOf(anchor, TextQuoteAnchor);
+      });
+    });
+
+    describe('#fromSelector', () => {
+      it('returns a TextQuoteSelector', () => {
+        const anchor = TextQuoteAnchor.fromSelector(container, {
+          exact: 'Liberty',
+          prefix: 'a new nation, conceived in ',
+          suffix: ', and dedicated to the proposition that',
+        });
+        assert.instanceOf(anchor, TextQuoteAnchor);
+      });
+    });
+
+    describe('#toSelector', () => {
+      it('returns a TextQuoteAnchor instance', () => {
+        const anchor = createTextQuoteAnchor();
+        assert.deepEqual(anchor.toSelector(), {
+          type: 'TextQuoteSelector',
+          exact: 'Liberty',
+          prefix: 'a new nation, conceived in ',
+          suffix: ', and dedicated to the proposition that',
+        });
+      });
+    });
+
+    describe('#toRange', () => {
+      it('returns a valid DOM Range', () => {
+        $imports.$restore();
```

Comment:
`$restore` undoes all of the active mocks. If for a particular test you just wanted to undo one mock, you can pass an argument to `$restore`. See https://github.com/robertknight/babel-plugin-mockable-imports#restoring-specific-mocks.

Reply:
good to know!